classes/publication/Repository.php

```diff
@@ -470,7 +485,12 @@ public function publish(Publication $publication)
                 $submission
             ]
         );
-        event(new PublishedEvent($newPublication, $publication, $submission));
+
+        $context = $submission->getData('contextId') === Application::get()->getRequest()->getContext()->getId()
```

Comment:
Since we're in the Repository class, far away from request handling, I think it's best not to assume we'll have a Request object (for a future where it's simply not available during non-interactive invocations). But this is an acceptable optimization, so I'd recommend using the null-safe operator:
```php
Application::get()->getRequest()?->getContext()?->getId()
```

Reply:
Good call.